flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java

```
@@ -133,6 +133,9 @@ public Long latestSnapshotId() {
             FileStatus[] statuses = fs.listStatus(snapshotDir);
 
             long latestId = Snapshot.FIRST_SNAPSHOT_ID - 1;
+            if (statuses == null) {
+                return null;
+            }
```

What will trigger this? When `snapshotDir` does not exist? If that is the case shall we throw an exception stating that we cannot find the specified snapshot directory?

---

I prefer to relax this restriction. read a non-existent table to return an empty collection. It is not necessary for writer to create the `snapshot` directory.

---

> What will trigger this? When snapshotDir does not exist? If that is the case shall we throw an exception stating that we cannot find the specified snapshot directory?

TestAtomicRenameFileSystem#listStatus

---

+1 for throw meaningful exception here

---

I think we can:
- add a debug log and return 0 for `file not exists`.
- Throw exception for `listStatus return null`.

---

I think we can:
- add a debug log and return 0 for `file not exists`.
- Throw exception for `listStatus return null`.